src/custom/mirage-gallery-curated/index.js

```diff
@@ -0,0 +1,98 @@
+import axios from "axios";
+import slugify from "slugify";
+
+function getProjectID(tokenId) {
+  let tokenStr = tokenId.toString();
+  if (tokenStr.length === 5) {
+    return parseInt(tokenStr[0], 10);
+  } else if (tokenStr.length === 6) {
+    return parseInt(tokenStr.slice(0, 2), 10);
+  } 
+}
+
+export const fetchCollection = async (_chainId, { contract, tokenId }) => {
+  
+  const projectID = getProjectID(tokenId);
+
+  const url = `https://account.miragegallery.ai/curated-details.json`;
+  const { data } = await axios.get(url);
+
+  const projectDetails = data.data.find(item => item.projectId === projectID);
+  const miragePayoutAddress = data.curatedPayoutAddress;
+  const mirageBPS = data.curatedPayoutBPS;
+  const artistBPS = data.artistPayoutBPS;
+
+  const startTokenId = tokenId - (tokenId % 10000);
+  const endTokenId = startTokenId + 10000 - 1;
+
+  let royalties;
+
+  if (projectDetails.secondaryArtistAddress == "") {
+    royalties = [{bps: mirageBPS, recipient: miragePayoutAddress}, {bps: artistBPS, recipient: projectDetails.artistAddress}]
+  } else {
+    royalties = [{bps: mirageBPS, recipient: miragePayoutAddress}, {bps: artistBPS / 2, recipient: projectDetails.artistAddress}, {bps: artistBPS / 2, recipient: projectDetails.secondaryArtistAddress}]
+  }
+
+  let extURL = projectDetails.website
+  if (projectDetails.website == "") {
+    extURL = "https://miragegallery.ai/curated"
+  }
+
+  return {
+    id: `${contract}:${startTokenId}:${endTokenId}`,
+    slug: slugify(projectDetails.name, { lower: true }),
+    name: projectDetails.dropName,
+    metadata: {
+      imageUrl: projectDetails.image,
+      bannerUrl: projectDetails.banner,
+      description: projectDetails.description,
+      externalUrl: extURL,
+    },
+    royalties,
```

Comment:
I was hoping that I could be fully non-reliant on OpenSea. OpenSea right now is having to manually sort our collections for us which ends up being a pain. 

The way that I'm doing it now, I'm adding the royalties directly from my site so that even if OpenSea was not up-to-date, it would still be returning the correct amounts and payout addresses.